### DISCUSSION TITLE
hoarder kills raspberry pi

### DISCUSSION THREAD

**Discussion by markusthiel** [karakeep-app/karakeep#263] [General]
I have tried using Hoarder to move away from Linkwarden. For this, I installed it via Docker on a Raspberry Pi 5, and it initially runs fine. However, with some added links, the system seems to struggle. Sometimes, it crashes the entire Raspberry Pi, and only a restart helps.

Does anyone have an idea what can be done about this?

I have a standard installation as indicated in the documentation, including the OpenAI connection.

**Comment by MohamedBassem** [comment]
That's kinda surprising specially that you're using openAI and not doing local inference. Chrome can be demanding, but I don't think it'd be demanding enough to crash the entire pi.

Maybe try adding some memory limits to the chrome container in particular (https://stackoverflow.com/questions/42345235/how-to-specify-memory-cpu-limit-in-docker-compose-version-3) and see if this can help?

**Comment by akshara-tg** [comment]
@markusthiel Did you check another users on your installation? I had forgotten to disable self-registration and one day when I had logged in to Admin user console saw there are lots of unknown accounts. Even I had the raspberry pi becoming unresponsive and I assume, these unauthorized users were doing some kind of brute force or large crawling.

@MohamedBassem 
I am trying to assess the extent of damage these unauthorized accounts created in my instance :-(
The point # 3 in this [link](https://docs.hoarder.app/security-considerations) is causing me sleepless nights! 
I have installed it using docker.
Can you please help me understand this # 3 more in detail?

**Reply by MohamedBassem** [reply]
@akshara-tg Sorry you had to go through this. The concern that is being highlighted in that link is imagine you have a service exposed at your local network but not to the internet (for example your router's admin page). Someone can add this to hoarder as a URL and Hoarder will crawl it for them and show them a screenshot of what's in there or the data that's returned.

Luckily, the worker container logs all the URLs it crawls, so you can check the logs to find out all the URLs that were crawled by those unknown accounts.
